packages/scaffold/src/client.ts

```diff
@@ -252,6 +258,28 @@ export class Web3ModalScaffold {
         document.body.insertAdjacentElement('beforeend', modal)
         resolve()
       })
+      const siweClient = this.getSiweClient()
+      if (siweClient?.options?.enabled) {
+        try {
+          const session = await siweClient.getSession()
+
+          AccountController.subscribe(async newAccountState => {
+            const { isConnected } = newAccountState
+            if (session && !isConnected) {
+              await siweClient.signOut()
+            } else if (isConnected && !session) {
+              ModalController.open({ view: 'ConnectingSiwe' })
+            }
+          })
+        } catch (error) {
+          // No session but wallet is connected
+          AccountController.subscribeKey('isConnected', isConnected => {
```

Comment:
Good catch on unsubscribing.
We need to subscribe because `AccountController.state.isConnected` is not reactive and is equal to `false` when we run those bits.